fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/interest/EndOfDayBalance.java

```diff
@@ -60,30 +63,66 @@ public BigDecimal cumulativeBalance(final BigDecimal interestToCompound) {
                 MoneyHelper.getRoundingMode());
     }
 
+    public void setDecimales(int decimales) {
+        this.decimales = decimales;
+    }
+
     public BigDecimal calculateInterestOnBalance(final BigDecimal interestToCompound, final BigDecimal interestRateAsFraction,
             final long daysInYear, final BigDecimal minBalanceForInterestCalculation, final BigDecimal overdraftInterestRateAsFraction,
             final BigDecimal minOverdraftForInterestCalculation) {
 
-        BigDecimal interest = BigDecimal.ZERO.setScale(9, MoneyHelper.getRoundingMode());
+        BigDecimal interest = BigDecimal.ZERO.setScale(this.decimales, MoneyHelper.getRoundingMode());
```

Comment:
This should not be a parameter, use MoneyHelper.PRECISION for calculations and Money for final amount